### PR TITLE
Added in-memory storing of inaccessible builds

### PR DESCRIPTION
**Pull request by mocenas**
added functionality to store builds, pushed via HTTP requests.
We will use it to display builds from jenkins, which is not accessible from dashboard.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -24,6 +24,7 @@
         <version.org.thymeleaf>3.0.9.RELEASE</version.org.thymeleaf>
         <version.org.thymeleaf.extras.java8time>3.0.1.RELEASE</version.org.thymeleaf.extras.java8time>
         <version.org.wildfly.swarm>2018.2.0</version.org.wildfly.swarm>
+        <version.apache.commins.lang>3.8.1</version.apache.commins.lang>
```
Comment (Ladicek) [inline pom.xml]: Typo: `commins` -> `common`. Not even mentioning that Apache Commons should generally be treated as obsolete crap.

File: src/main/java/com/github/rhoar_ci/dashboard/jenkins/JsonJobDataInDescription.java [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+
+public class JsonJobDataInDescription {
+    public String cluster;
+    public String description;
+    public String type;
+
+    public static Optional<JsonJobDataInDescription> parse(String inputDescription){
+        try {
+            String description = inputDescription.replaceAll("<!-- .*? -->", "");
+            return Optional.of((new Gson()).fromJson(description, JsonJobDataInDescription.class));
```
Comment (Ladicek) [inline src/main/java/com/github/rhoar_ci/dashboard/jenkins/JsonJobDataInDescription.java]: Why the extra parens around `new Gson()`?

File: src/main/java/com/github/rhoar_ci/dashboard/storage/JobService.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import com.github.rhoar_ci.dashboard.ci.TestDescription;
+import com.github.rhoar_ci.dashboard.jenkins.Job;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.enterprise.context.RequestScoped;
+import java.util.*;
+
+@RequestScoped
+public class JobService {
+    private static Map<Pair<TestCluster, TestDescription>, Job> jobStorage= new HashMap<>();
```
Comment (Ladicek) [inline src/main/java/com/github/rhoar_ci/dashboard/storage/JobService.java]: Why not make it an `@ApplicationScoped` bean and this non-`static`?

File: src/main/java/com/github/rhoar_ci/dashboard/storage/JobService.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    public void updateJob(Job jenkinsJob){
+        Pair<TestCluster, TestDescription> pair= jobToPair(jenkinsJob);
```
Comment (Ladicek) [inline src/main/java/com/github/rhoar_ci/dashboard/storage/JobService.java]: Space before `=`.

**Review by Ladicek — CHANGES_REQUESTED** [review]

**Comment by Ladicek** [comment]
Also a big problem is that the additional builds are only stored in memory. When OpenShift decides to restart the pod, it's all lost.

**Comment by mocenas** [comment]
@Ladicek thanks for comments, it should be resolved.
I know that storing this just in-memory might be a problem. But we've decided that it is acceptable. Displaying these builds is not critical, and the pod seems to have a pretty long live time.

**Comment by Ladicek** [comment]
Hmm, OK then...

**Pull request merged**